app/main.py

```diff
@@ -1,3 +1,48 @@
-def shop_trip():
-    # write your code here
-    pass
+from app.customer import Customer
+from app.shop import Shop
+from app.utils import load_config
+
+
+def shop_trip() -> None:
+    config = load_config("config.json")
+    fuel_price = config["FUEL_PRICE"]
+    customers_data = config["customers"]
+    shops_data = config["shops"]
+
+    customers = [Customer(**data) for data in customers_data]
+    shops = [Shop(**data) for data in shops_data]
+
+    for customer in customers:
+        print(f"{customer.name} has {customer.money} dollars")
+        min_cost = float("inf")
+        best_shop = None
+        for shop in shops:
+            trip_cost = customer.calculate_trip_cost(shop, fuel_price)
+            print(
+                f"{customer.name}'s trip to the {shop.name} "
+                f"costs {trip_cost:.2f}"  # noqa: E231
+            )  # noqa: E231
+            if trip_cost < min_cost:
+                min_cost = trip_cost
+                best_shop = shop
+
+        if best_shop and min_cost <= customer.money:
+            print(f"{customer.name} rides to {best_shop.name}")
```

Comment:
when customer goes to shop his location should be changed to shop location, same when he arrives home